### DISCUSSION TITLE
MacOS Menubar/TrayIcon Icon Click handling

### DISCUSSION THREAD

**Discussion by josh-unwin** [AvaloniaUI/Avalonia#16333] [Q&A]
Hi all,

Has anyone found a workaround to implement Click handling of a Menubar/Tray icon in MacOS?
Applying a `Clicked` parameter to `<TrayIcon>` is unsupported on MacOS as per the source here:
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/TrayIcon.cs
```
        /// <summary>
        /// Raised when the TrayIcon is clicked.
        /// Note, this is only supported on Win32 and some Linux DEs, 
        /// on OSX this event is not raised.
        /// </summary>
        public event EventHandler? Clicked;
```

I saw someone seemed to make a workaround back in 2021 in this thread but it's using an old tool (MonoMac), is there a better way at this point?
https://github.com/AvaloniaUI/Avalonia/issues/2649

Many thanks

**Comment by usingsystem007** [comment]
I completely agree - this functionality is essential and should definitely be prioritized.
